src/Common/src/Interop/Unix/System.Net.Security.Native/SecuritySafeHandles.cs

```diff
@@ -13,16 +13,65 @@ namespace System.Net.Security
     internal sealed class SafeFreeNegoCredentials : SafeFreeCredentials
     {
         private SafeGssCredHandle _credential;
+        private readonly bool _isNtlm;
+        private readonly string _userName;
+        private readonly bool _isDefault;
 
         public SafeGssCredHandle GssCredential
         {
             get { return _credential; }
         }
 
-        public SafeFreeNegoCredentials(string username, string password, string domain) : base(IntPtr.Zero, true)
+        public bool IsNtlm
         {
+            get { return _isNtlm; }
+        }
+
+        public string UserName
+        {
+            get { return _userName; }
+        }
+
+        public bool IsDefault
+        {
+            get { return _isDefault; }
+        }
+
+        public SafeFreeNegoCredentials(bool isNtlm, string username, string password, string domain)
+            : base(IntPtr.Zero, true)
+        {
+            Debug.Assert(username != null && password != null, "Username and Password can not be null");
+            const char At = '@';
+            const char Backwhack = '\\';
+
+            int index = username.IndexOf(Backwhack);
+            if (index > 0 && username.IndexOf(Backwhack, index + 1) < 0 && string.IsNullOrEmpty(domain))
```

Comment:
yes, that's the intention . Any invalid format passed will be taken as it is and no manipulation should be done